src/module/mod.rs

```diff
@@ -101,7 +104,15 @@ impl Module {
                     // if it comes out here then it means the section may be
                     // malformed according to parity-wasm, so skip it.
                     if s.name() == "name" {
-                        continue
+                        continue;
+                    }
+                    if s.name() == "name" {
```

Comment:
bug? `s.name() == "name"` is handled just above. Should this be `s.name() == "producers"`?